### PR TITLE
[CHK 2096] feat: generate unique orderId with prefix and cache check

### PR DESCRIPTION
**Pull request by Fernando-Granato**
depend on https://github.com/pagopa/pagopa-ecommerce-commons/pull/152
depend on https://github.com/pagopa/pagopa-ecommerce-local/pull/106

#### List of Changes
The change was made to ensure the uniqueness of the generation of the orderId to be forwarded to NPG.
In particular, a logic was defined that saves to redis to check that the generated orderId does not collide with other previously received requests.
A prefix was added to the generation of the unique string. **(E)**
The duration of the id in the cache is set to one minute when the collision could occur to the millisecond.

#### Motivation and Context
The change was made to generate a unique orderId.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -229,6 +229,8 @@
 					<connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
 					<scmVersionType>tag</scmVersionType>
 					<scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+					<scmVersionType>branch</scmVersionType>
+					<scmVersion>CHK-2095-NX-redisTemplateWrapper</scmVersion>
```
Comment (Fernando-Granato) [inline pom.xml:233]:
```suggestion

```

Comment (Fernando-Granato) [inline pom.xml:233]:
```suggestion
```

**Review by pietro-tota — DISMISSED** [review]
LGMT 🚀

**Review by ciuffagianluca — APPROVED** [review]

File: src/main/java/it/pagopa/ecommerce/payment/methods/utils/UniqueIdUtils.java [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
+import java.time.Duration;
 
 @Component
 public class UniqueIdUtils {
+    private final UniqueIdTemplateWrapper uniqueIdTemplateWrapper;
     private static final SecureRandom secureRandom = new SecureRandom();
     private static final String ALPHANUMERICS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._";
     private static final int MAX_LENGTH = 18;
+    private static final int MAX_NUMBER_ATTEMPTS = 3;
+    private static final String PRODUCT_PREFIX = "E";
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/payment/methods/utils/UniqueIdUtils.java:20]: I would make these package-private so they can be used in tests

```suggestion
    static final int MAX_NUMBER_ATTEMPTS = 3;
    static final String PRODUCT_PREFIX = "E";
```

**Review by giovanniberti — DISMISSED** [review]

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**